sendgrid-client/src/main/java/uk/gov/hmcts/reform/sendgrid/SendGridClient.java

```diff
@@ -0,0 +1,123 @@
+package uk.gov.hmcts.reform.sendgrid;
+
+import com.sendgrid.Method;
+import com.sendgrid.Request;
+import com.sendgrid.Response;
+import com.sendgrid.SendGrid;
+import com.sendgrid.helpers.mail.Mail;
+import com.sendgrid.helpers.mail.objects.Attachments;
+import com.sendgrid.helpers.mail.objects.Content;
+import com.sendgrid.helpers.mail.objects.Email;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpException;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Recover;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class SendGridClient {
+
+    private static final String TEXT_PLAIN_VALUE = "text/plain";
+
+    private final SendGrid sendGrid;
+
+    @Retryable(value = EmailSendFailedException.class, backoff = @Backoff(delay = 100, maxDelay = 500))
+    public void sendEmail(String from, EmailData emailData) {
+        verifyData(from, emailData);
+        try {
+            Email sender = new Email(from);
+            String subject = emailData.getSubject();
+            Email recipient = new Email(emailData.getTo());
+            Content content = new Content(TEXT_PLAIN_VALUE, getMessage(emailData));
+            Mail mail = new Mail(sender, subject, recipient, content);
+            if (emailData.hasAttachments()) {
+                emailData.getAttachments().stream()
+                    .map(SendGridClient::toSendGridAttachments)
+                    .forEach(mail::addAttachments);
+            }
+
+            Request request = new Request();
+            request.setMethod(Method.POST);
+            request.setEndpoint("mail/send");
+            request.setBody(mail.build());
+
+            Response response = sendGrid.api(request);
+            if (!is2xxSuccessful(response)) {
+                throw new EmailSendFailedException(new HttpException(String.format(
+                    "SendGrid returned a non-success response (%d); body: %s",
+                    response.getStatusCode(),
+                    response.getBody()
+                )));
+            }
+        } catch (IOException exception) {
+            throw new EmailSendFailedException(exception);
+        }
+    }
+
+    private boolean is2xxSuccessful(Response response) {
+        return response.getStatusCode() / 100 == 2;
+    }
```

Comment:
is there not a method under HttpStatus for this?

Reply:
Do not want to add the dependency for just little this check.